packages/wallets/src/evm/connectors/smart-wallet/lib/http-rpc-client.ts

```diff
@@ -2,8 +2,12 @@ import { providers, utils } from "ethers";
 import { UserOperationStruct } from "@account-abstraction/contracts";
 import { deepHexlify } from "@account-abstraction/utils";
 import { isTwUrl } from "../../../utils/url";
+import pkg from "../../../../../package.json";
 
 const DEBUG = false;
+function isBrowser() {
+  return typeof window !== "undefined";
```

Comment:
This is too simple imo, platforms like ReactNative will also return true here. I recently discovered that this is a thing:

`global?.navigator?.product === 'ReactNative'`, so we can use this to differentiate react-native, browser, node

Reply:
Ooh that's nice, can you add it for me?

Def want to differentiate reqs from react native, thanks for catching that

Reply:
hmmm that gives me weird names in browser though. I think i'll just consider react native if we have a bundle id

Reply:
Yeah, looking deeper it seems that the feature is deprecated: 

https://developer.mozilla.org/en-US/docs/Web/API/Navigator/product

Reply:
It seems we can do something like this though:

```
function getEnvironment() {
  if (typeof navigator !== 'undefined' && navigator.product === 'ReactNative') {
    return 'React Native';
  } else if (typeof window !== 'undefined') {
    return 'React';
  } else {
    return 'Unknown';
  }
}
```